atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleAlterTableTest.java

```diff
@@ -0,0 +1,180 @@
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleTableNameGetter;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleTableNameGetterImpl;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.ConnectionManagerAwareDbKvs;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.ConnectionSupplier;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.OverflowMigrationState;
+import com.palantir.atlasdb.keyvalue.impl.TestResourceManager;
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.common.exception.TableMappingNotFoundException;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class OracleAlterTableTest {
+
+    private static final Namespace NAMESPACE = Namespace.create("test_namespace");
+    private static final TableReference TABLE_REFERENCE = TableReference.create(NAMESPACE, "foo");
+
+    private static final DbKeyValueServiceConfig CONFIG = ImmutableDbKeyValueServiceConfig.builder()
+            .from(DbKvsOracleTestSuite.getKvsConfig())
+            .ddl(ImmutableOracleDdlConfig.builder()
+                    .overflowMigrationState(OverflowMigrationState.FINISHED)
+                    .build())
+            .build();
+
+    private static final DbKeyValueServiceConfig CONFIG_WITH_ALTER = ImmutableDbKeyValueServiceConfig.builder()
+            .from(CONFIG)
+            .ddl(ImmutableOracleDdlConfig.builder()
+                    .overflowMigrationState(OverflowMigrationState.FINISHED)
+                    .addAlterTablesOrMetadataToMatch(TABLE_REFERENCE)
+                    .build())
+            .build();
+
+    @ClassRule
+    public static final TestResourceManager TRM =
+            new TestResourceManager(() -> ConnectionManagerAwareDbKvs.create(CONFIG));
+
+    private static final String COLUMN_NAME = "variable";
+
+    private static final TableMetadata EXPECTED_TABLE_METADATA = TableMetadata.builder()
+            .singleRowComponent("name", ValueType.STRING)
+            .singleNamedColumn("var", COLUMN_NAME, ValueType.STRING)
+            .build();
+
+    private static final TableMetadata OLD_TABLE_METADATA = TableMetadata.builder()
+            .singleRowComponent("name", ValueType.STRING)
+            .singleNamedColumn("var", COLUMN_NAME, ValueType.FIXED_LONG)
+            .build();
+
+    private static final Cell DEFAULT_CELL = createCell("foo", "bar");
+    private static final Cell DEFAULT_CELL_2 = createCell("something", "bar");
+
+    private static final byte[] DEFAULT_VALUE = PtBytes.toBytes("amazing");
+    private static final long TIMESTAMP = 1L;
+
+    private ConnectionManagerAwareDbKvs defaultKvs;
+    private ConnectionSupplier connectionSupplier;
+    private OracleTableNameGetter oracleTableNameGetter;
+
+    @Before
+    public void setup() {
+        defaultKvs = (ConnectionManagerAwareDbKvs) TRM.getDefaultKvs();
```

Comment:
I reckon you can get away with just KeyValueService as the type (without the cast, and using connection manager to get connections - see OracleNamespaceDeleterIntegrationTest)

Reply:
Also, we generally call our methods before() and after()

Reply:
I ripped these from another test, which is why it's setup()/teardown(), but happy to go with before() after()!

Reply:
As for (1), you're correct! Much cleaner.